rules/src/main/scala/fix/GenerateDataClass.scala

```diff
@@ -183,7 +183,7 @@ class GenerateDataClass(config: Configuration) extends SemanticRule("GenerateDat
     def cleanParam(param: Term.Param): Term.Param =
       param.copy(
         default = None,
-        mods = param.mods.filterNot(m => m.toString.startsWith(sinceStr)),
+        mods = Nil,
```

Comment:
why was this removed?

Reply:
Let's say someone writes this:

```scala
@data class Foo(@since private final val bar: Int)
```

When `scalafix` generates the code, we should get this:

```scala
object Foo {
  def apply(bar: Int): Foo = new Foo(bar)
}
```

Not only `Mod.Annot` should be removed but the other modifiers too. The original motivation was to remove `Mod.ValParam`, but the others can't be in the `apply` method either.